### ISSUE TITLE
querySelectorAllExt returns unexpected null which impacts findAttributeTargets

### ISSUE THREAD

**Issue by san4d** [bigskysoftware/htmx#3055]
## Summary
`querySelectorAllExt` can return `[null]` which violates the documented return type of `findAttributeTargets`: [(Node|Window)[]](https://github.com/bigskysoftware/htmx/blob/master/src/htmx.js#L1303). This has upstream effects on methods like [disableElements](https://github.com/bigskysoftware/htmx/blob/master/src/htmx.js#L3260), which iterate through a collection of elements returned by `findAttributeTargets`, expecting each to be not null.

```javascript
  function findAttributeTargets(elt, attrName) {
    const attrTarget = getClosestAttributeValue(elt, attrName)
    if (attrTarget) {
      if (attrTarget === 'this') {
        return [findThisElement(elt, attrName)]
      } else {
        const result = querySelectorAllExt(elt, attrTarget) // <-- This can return [null]
        if (result.length === 0) {
          logError('The selector "' + attrTarget + '" on ' + attrName + ' returned no matches!')
          return [DUMMY_ELT]
        } else {
          return result
        }
      }
    }
  }
```

## Context
In my application, this happens when the `elt` passed in to `findAttributeTargets` has no children and the `attrTarget` would yield no results. The `attrName` in my context is `hx-disabled-elt`. 

## Workaround
In my setup, the inherited `hx-disabled-elt` was causing the failed lookup. I explicitly set the `hx-disabled-elt` to a query selector that would return results to unblock myself.

**Comment by MichaelWest22** [comment]
I think this was found here #2913 which i wrote a quick fix for but this fix has now been combined with an existing in progress enhancment PR #2902 which should resolve this when this is merged and released.

**Comment by san4d** [comment]
Thanks, that looks similar. I'll close this for now, follow 2902, and retest once that's merged.

**Comment by san4d** [comment]
Closing in favor of https://github.com/bigskysoftware/htmx/pull/2902.
